addons/sms/static/src/core/message_notification_popover_patch.js

```diff
@@ -0,0 +1,27 @@
+import { MessageNotificationPopover } from "@mail/core/common/message_notification_popover";
+import { patch } from "@web/core/utils/patch";
+import { useService } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
+
+patch(MessageNotificationPopover.prototype, {
```

Comment:
Why not take example on the snailmail model and patch message instead of the popover. If I am not mistaken you have access to the failure_type of each notification. In the patch you override `onClickNotification` to directly open the view if we are in the correct situation. Then no need for this one and, maybe, the change you did in snailmail.